src/darwin/Framework/CHIP/MTROTAProviderDelegateBridge.mm

```diff
@@ -597,9 +661,16 @@ bool GetPeerNodeInfo(CommandHandler * commandHandler, const ConcreteCommandPath
                           handle.Release();
                           return;
                       }
-
-                      handler->AddResponse(cachedCommandPath, response);
+                      if (!isBDXProtocolSupported) {
+                          Commands::QueryImageResponse::Type protocolNotSupportedResponse;
+                          protocolNotSupportedResponse.status
+                              = static_cast<OTAQueryStatus>(MTROTASoftwareUpdateProviderOTAQueryStatusDownloadProtocolNotSupported);
+                          handler->AddResponse(cachedCommandPath, protocolNotSupportedResponse);
+                      } else {
+                          handler->AddResponse(cachedCommandPath, response);
+                      }
```

Comment:
So I just realized this logic is not right.  If we get here, that means that at least one of `isBDXProtocolSupported` and `hasUpdate` is false.  If `hasUpdate` is false we really need to send the exact `response` that we have right now.  So this should be more like:
```suggestion
                      if (!hasUpdate) {
                          // Send whatever error response our delegate decided on.
                          handler->AddResponse(cachedCommandPath, response);
                      } else {
                          // We must have isBDXProtocolSupported false.  Send the corresponding error status.
                          Commands::QueryImageResponse::Type protocolNotSupportedResponse;
                          protocolNotSupportedResponse.status
                              = static_cast<OTAQueryStatus>(MTROTASoftwareUpdateProviderOTAQueryStatusDownloadProtocolNotSupported);
                          handler->AddResponse(cachedCommandPath, protocolNotSupportedResponse);
                      }
```